docs/datamodel/properties.rst

```diff
@@ -0,0 +1,200 @@
+
+    # an unordered set of strings
+    multi property nicknames -> str;
+
+    # an unordered set of string arrays
+    multi property set_of_arrays -> array<str>;
+  }
+
+**Comparison to arrays**
+
+The values associated with a ``multi`` property are stored in no
+particular order. If order is important, use an :ref:`array
+<ref_datamodel_arrays>`. Otherwise, ``multi`` properties are recommended. For a
+more involved discussion, see :ref:`EdgeQL > Sets
+<ref_eql_set_array_conversion>`.
+
+Default values
+--------------
+
+Properties can have a default value. This default can be a static value or an
+arbitrary EdgeQL expression, which will be evaluated upon insertion.
+
+.. code-block:: sdl
+
+  type Player {
+    required property points -> int64 {
+      default := 0;
+    }
+
+    required property latitude -> float64 {
+      default := (360 * random() - 180);
+    }
+  }
+
+Readonly properties
+-------------------
+
+Properties can be marked as ``readonly``. In the example below, the
+``User.external_id`` property can be set at the time of creation but not
+modified thereafter.
+
+.. code-block:: sdl
+
+  type User {
+    required property external_id -> uuid {
+      readonly := true;
+    }
+  }
+
+
+Constraints
+-----------
+
+Properties can be augmented wth constraints. The example below showcases a
+subset of EdgeDB's built-in constraints.
+
+.. code-block:: sdl
+
+  type BlogPost {
+    property title -> str {
+      constraint exclusive; # all post titles must be unique
+      constraint min_len_value(8);
+      constraint max_len_value(30);
+      constraint regexp(r'^[A-Za-z0-9 ]+$');
+    }
+
+    property status -> str {
+      constraint one_of('Draft', 'InReview', 'Published');
+    }
+
+    property upvotes -> int64 {
+      constraint min_value(0);
+      constraint max_value(9999);
+    }
+  }
+
+You can constrain properties with arbitrary :ref:`EdgeQL <ref_edgeql>`
+expressions returning ``bool``. To reference to value of the property, use the
+special scoped keyword ``__subject__``.
+
+.. code-block:: sdl
+
+  type BlogPost {
+    property title -> str {
+      constraint expression on (
+        __subject__ = str_trim(__subject__)
+      );
+    }
+  }
+
+The constraint above guarantees that ``BlogPost.title`` doesn't contain any
+leading or trailing whitespace by checking that the raw string is equal to the
+trimmed version. It uses the built-in :eql:func:`str_trim` function.
+
+For a full reference of built-in constraints, see the :ref:`Constraints
+reference <ref_std_constraints>`.
+
+
+Annotations
+-----------
+
+Properties can contain annotations, small human-readable notes. The built-in
+annotations are ``title``, ``description``, and ``deprecated``. You can declare
+:ref:`custom annotation types <ref_datamodel_inheritance_annotations>`.
+
+.. code-block:: sdl
+
+  type User {
+    property email -> str {
+      annotation title := 'Email address';
+      annotation description := 'The user\'s email address.';
```

Comment:
```suggestion
      annotation description := "The user's email address.";
```
Please use `"The user's email address."` in the example. I understand that when we return strings (including `describe`) we will display them as you've shown here, but in your own code you can and often should use quotes that avoid escaping.